src/v/crypto/key.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/bytes.h"
+#include "crypto/crypto.h"
+#include "crypto/types.h"
+#include "ssl_utils.h"
+
+namespace crypto {
+class key::impl {
```

Comment:
Would it make sense to put this under a `detail` namespace if consumers are all library-internal? 

Or maybe the idea is that this header wouldn't be exported at all?